datalad/distribution/uninstall.py

```diff
@@ -104,106 +106,85 @@ class Uninstall(Interface):
             args=("--data-only",),
             doc="if set, only data is uninstalled, but the handles are kept",
             action="store_true"),
-        recursive=Parameter(
-            args=("-r", "--recursive"),
-            doc="""if set, uninstall recursively, including all subdatasets.
-            The value of `data` is used for recursive uninstallation, too""",
-            action="store_true"),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
         fast=Parameter(
             args=("--fast",),
             doc="when uninstalling (sub-)datasets, don't try uninstalling its "
                 "data first. Warning: This will silently ignore any issue "
                 "regarding the uninstallation of contained data.",
-            action="store_true",))
+            action="store_true",),
+        if_dirty=if_dirty_opt,
+    )
 
     @staticmethod
     @datasetmethod(name='uninstall')
-    def __call__(path=None, dataset=None, data_only=False, recursive=False,
-                 fast=False):
+    def __call__(
+            path=None,
+            dataset=None,
+            data_only=False,
+            recursive=False,
+            recursion_limit=None,
+            fast=False,
+            if_dirty='save-before'):
```

Comment:
I am not sure. ATM I would find it technically complicated to do that, and I am not sure of there is always a single default that works for any command. If not, you'd need to look in the same spots anyway.